fixed export when called without arguments

Vinni-Cedraz opened the pull request:
Also made get_index return the correct index without having to add + 1 to it. But this revealed an issue with the way cursor->token is being initialized, because it is receiving a number one smaller than it should.

	modified:   include/minishell.h
	modified:   src/executor/executor.c
	modified:   src/syntax/analysis.c
	modified:   src/syntax/lexer_strings.c
	modified:   src/utils/find_token.c

v-ribeiro-tixit committed:
fixed export when called without arguments

Also made get_index return the correct index without having to add + 1
to it. But this revealed an issue with the way cursor->token is being
initialized, because it is receiving a number one smaller than it
should.

	modified:   include/minishell.h
	modified:   src/executor/executor.c
	modified:   src/syntax/analysis.c
	modified:   src/syntax/lexer_strings.c
	modified:   src/utils/find_token.c

v-ribeiro-tixit committed:
merged with fine_tuning...

astahjmo commented:
Build error, fix it so we can merge 
```
./src/builtins/export_builtin.c:16:17: error: static declaration of ‘is_valid_env_name’ follows non-static declaration
   16 | static int      is_valid_env_name(char *env);
      |                 ^~~~~~~~~~~~~~~~~
In file included from ./src/builtins/export_builtin.c:13:
./include/minishell.h:165:49: note: previous declaration of ‘is_valid_env_name’ with type ‘int(char *)’
  165 | int                                             is_valid_env_name(char *name_value);
      |                                                 ^~~~~~~~~~~~~~~~~
make: *** [Makefile:53: objs/export_builtin.o] Error 1
````

v-ribeiro-tixit committed:
improve export to behave more like bash

if you export a variable that already exists (and initialize its value),
it should overwrite the previous one, wherever it is found within env.

	modified:   include/minishell.h
	modified:   src/builtins/export_builtin.c
	modified:   src/builtins/utils_builtins.c

(amend): just cleaning binary files from previous commit
	modified:   include/minishell.h
	modified:   src/builtins/export_builtin.c
	modified:   src/builtins/utils_builtins.c

v-ribeiro-tixit committed:
making export and env more bash_like

-> export is able to create envars without value
-> env should not display envars without value
-> export without arguments should display even variables without value
with "declare -x" as its prefix

	modified:   Makefile
	modified:   include/minishell.h
	modified:   src/builtins/env_builtin.c
	modified:   src/builtins/export_builtin.c
	modified:   src/builtins/utils_builtins.c
	deleted:    src/utils/get_env_content.c

v-ribeiro-tixit committed:
make export behave more like in bash

export without arguments should display all initialized variables (even
those initialized with an empty string) with double quotes around it's
content
    --> and this means uninitialized ones should have neither content
    nor any quotes around anything

	modified:   src/builtins/env_builtin.c

v-ribeiro-tixit committed:
(env when variable initialized with empty str)

prints NAME=
and when uninitialized keeps not printing anything

	modified:   include/minishell.h
	modified:   src/builtins/env_builtin.c
	modified:   src/builtins/utils_builtins.c

v-ribeiro-tixit committed:
get_env_content is not in a loop anymore

and it is bzeroing content only when it really has to do so
	modified:   include/minishell.h
	modified:   src/builtins/env_builtin.c
	modified:   src/builtins/utils_builtins.c

v-ribeiro-tixit committed:
get_env_name is not a loop anymore

	modified:   include/minishell.h
	modified:   src/builtins/env_builtin.c
	modified:   src/builtins/utils_builtins.c

v-ribeiro-tixit committed:
unset is working with multiple arguments

and no memory leaks caused by it
	modified:   src/builtins/unset_builtin.c

Vinni-Cedraz committed:
Merge branch 'dev' into fix_no_argument_export
